pkg/loop/adapters/relay/adapter.go

```diff
@@ -22,6 +22,10 @@ type RelayerAdapter struct {
 	RelayerExt
 }
 
+func (r *RelayerAdapter) NewContractReader(_ context.Context, contractReaderConfig []byte) (types.ChainReader, error) {
```

Comment:
Would it be worth aliasing a type? Gives some flexibility for now while keeping the naming aligned locally:
```suggestion
func (r *RelayerAdapter) NewContractReader(_ context.Context, contractReaderConfig []byte) (types.ContractReader, error) {
```

Reply:
Changed